kernel/src/sync/semaphore.rs

```diff
@@ -61,6 +64,52 @@ impl Semaphore {
         self.acquire();
         SemaphoreGuard { sem: self }
     }
+
+    pub fn check(&self, k: isize) -> Result<bool, ()> {
+        let mut count = self.lock.lock();
+        Ok((*count) >= k)
+    }
+
+    pub fn get(&self) -> Result<isize, ()> {
+        let mut count = self.lock.lock();
+        Ok((*count))
+    }
+
+    pub fn set(&self, k: isize) -> Result<(), ()> {
+        let mut count = self.lock.lock();
+        *count = k;
+        Ok(())
+    }
+
+    /// Modify by k atomically. when wait is false avoid waiting. unused
+    pub fn modify(&self, k: isize, wait: bool) -> Result<usize, ()> {
+        match (k) {
```

Comment:
Do not need `()` in conditions in Rust.
```suggestion
        match k {
```